packages/std/src/math.rs

```diff
@@ -89,18 +90,35 @@ impl FromStr for Decimal {
                     .ok_or_else(|| generic_err("Cannot compute fractional factor"))?;
 
                 let whole_as_atomics = whole
-                    .checked_mul(DECIMAL_FRACTIONAL.u128())
+                    .checked_mul(DECIMAL_FRACTIONAL)
```

Comment:
all these `checked_add`, `checked_mul` is cool stuff.
Very nice parse logic here.